corehq/apps/data_interfaces/utils.py

```diff
@@ -214,6 +215,23 @@ def _get_sql_repeat_record(domain, record_id):
         return None
 
 
+def _throttle_case_iterator(case_iterator):
+    """
+    Throttles slower queries that hog pgshard resources by sleeping for the amount of time the query took. Used
+    only in iter_cases_and_run_rules to reduce load on db when running through all update rules.
+    We currently aren't able to iterate through all cases on the largest domains within the alloted time,
+    so this .
+    """
+    start_time = time.time()
+    for case in case_iterator:
+        end_time = time.time()
+        query_time = end_time - start_time
+        if query_time > 0.08:
+            time.sleep(query_time)
```

Comment:
I think I still like
```suggestion
            time.sleep(query_time - 0.08)
```
or
```suggestion
            time.sleep(2 * (query_time - 0.08))
```

better, because there's no discontinuity in wait times. The way you have it here, wait time is at 80ms is 0ms and at 81 it's 81ms—big jump.

Reply:
Great point. Updated in 6488628ba0a41a95a689a160da385adf6f2d706b